### ISSUE TITLE
Auto block/ban users with too many invalid login attempts

### ISSUE THREAD

**Issue by SDPrio** [FriendsOfSymfony/FOSUserBundle#2168]
I am using FOSUserBundle in Symfony 2.7. Is there any why of automatically block or ban users with too many invalid login attempts? The Goal is to prevent brute force attacks. 

I am aware of the user security features build into Symfony and FOSUserBundle. However such a filter would be an additional hurdle attackers would have to take.

I have been using WordPress before. There exist quite a lot of plugins and tools to automatically recognize and handle such attacks: If denied login attempts reaches a certain threshold, the user/ip is automatically blocked for some time. Usually after a few minutes the user/ip is automatically removed from the block list.

I have not been able to find such a solution for Symfony. Is there any bundle that integrates these functions into Symfony?

Of course it would not be too difficult to implement this functionally on my own. But it makes no sense to re-invent something that is already out there.

**Comment by stof** [comment]
This is out of the scope of FOSUserBundle IMO. It should be handled by a dedicated bundle.

I saw https://github.com/anyx/LoginGateBundle implementing such feature btw (and I made them some suggestion to improve their implementation). They block only by IP though.
Note that blocking by username alone would open the door to DoS attacks, as it would allow anyone to lock the account of another user just by making failed login attempts until the threshold is reached.
